test/e2e/acl_logging.go

```diff
@@ -0,0 +1,204 @@
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	knet "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	logSeverityNamespaceAnnotation = "k8s.ovn.org/acl-logging"
+	maxPokeRetries                 = 15
+	ovnControllerLogPath           = "/var/log/openvswitch/ovn-controller.log"
+	pokeInterval                   = 1 * time.Second
+)
+
+var _ = Describe("ACL Logging", func() {
+	const (
+		denyAllPolicyName       = "default-deny-all"
+		initialDenyACLSeverity  = "alert"
+		initialAllowACLSeverity = "notice"
+		namespacePrefix         = "acl-logging"
+		pokerPodIndex           = 0
+		pokedPodIndex           = 1
+	)
+
+	fr := framework.NewDefaultFramework(namespacePrefix)
+
+	var (
+		nsName string
+		pods   []v1.Pod
+	)
+
+	setNamespaceACLLogSeverity := func(namespaceToUpdate *v1.Namespace, desiredDenyLogLevel string, desiredAllowLogLevel string) error {
+		if namespaceToUpdate.ObjectMeta.Annotations == nil {
+			namespaceToUpdate.ObjectMeta.Annotations = map[string]string{}
+		}
+		By("updating the namespace's ACL logging severity")
+		updatedLogSeverity := fmt.Sprintf(`{ "deny": "%s", "allow": "%s" }`, desiredDenyLogLevel, desiredAllowLogLevel)
+		namespaceToUpdate.Annotations[logSeverityNamespaceAnnotation] = updatedLogSeverity
+
+		_, err := fr.ClientSet.CoreV1().Namespaces().Update(context.TODO(), namespaceToUpdate, metav1.UpdateOptions{})
+		return err
+	}
+
+	BeforeEach(func() {
+		By("configuring the ACL logging level within the namespace")
+		nsName = fr.Namespace.Name
+		namespace, err := fr.ClientSet.CoreV1().Namespaces().Get(context.Background(), nsName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "failed to retrieve the namespace")
+		Expect(setNamespaceACLLogSeverity(namespace, initialDenyACLSeverity, initialAllowACLSeverity)).To(Succeed())
+	})
+
+	BeforeEach(func() {
+		By("creating a \"default deny\" network policy")
+		_, err := makeDenyAllPolicy(fr, nsName, denyAllPolicyName)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	BeforeEach(func() {
+		By("creating pods")
+		cmd := []string{"/bin/bash", "-c", "/agnhost netexec --http-port 8000"}
+		for i := 0; i < 2; i++ {
+			pod := newAgnhostPod(fmt.Sprintf("pod%d", i+1), cmd...)
+			pod = fr.PodClient().CreateSync(pod)
+			Expect(waitForACLLoggingPod(fr, nsName, pod.GetName())).To(Succeed())
+			pods = append(pods, *pod)
+		}
+	})
+
+	BeforeEach(func() {
+		By("sending traffic between acl-logging test pods we trigger ACL logging")
+
+		clientPod := pods[pokerPodIndex]
+		pokedPod := pods[pokedPodIndex]
+		framework.Logf(
+			"Poke pod %s (on node %s) from pod %s (on node %s)",
+			pokedPod.GetName(),
+			pokedPod.Spec.NodeName,
+			clientPod.GetName(),
+			clientPod.Spec.NodeName)
+		Expect(
+			pokePod(fr, clientPod.GetName(), pokedPod.Status.PodIP)).To(HaveOccurred(),
+			"traffic should be blocked since we only use a deny all traffic policy")
+	})
+
+	AfterEach(func() {
```

Comment:
Why do we need this? Won't the test finishing automatically delete the namespace and everything with it?

Reply:
Usually If there is a var that needs to be redefined each test I would just re-declare it in each test, but I'm fine with this solution as well